src/Snap/Internal/Core.hs

```diff
@@ -767,7 +767,7 @@ dir :: MonadSnap m
     -> m a
 dir = pathWith f
   where
-    f dr pinfo = dr == x
+    f dr pinfo = dr == x || (dr == "" || dr == "/" || dr == ".")
```

Comment:
Don't think this patch will work, pathWith works on the length of "dr" and as such you'll get incorrect results here. Not only that, but the intention of "dir" is to move the context one level down into the path hierarchy based on an exact match of the prefix. Seems to me like `dir "."` should really be just `id`? What's the use case?